File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini = INT_MAX;
        int maxi = 0;
        int ans = 0;
        for(int i=0; i<prices.size(); i++){
            if(prices[i] < mini){
                mini = prices[i];
            }
            else {
                maxi = prices[i];
                ans = max(ans, (maxi - mini) );
            }
        }
        return ans;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini = Integer.MAX_VALUE;
        int maxi = 0;
        for(int i=0; i<prices.length; i++){
            if(prices[i] < mini){
                mini = prices[i];
            }
            else {
                maxi = Math.max(prices[i] - mini, maxi);
            }
        }
        return maxi;
    }
}